shared/chat/conversation/list/container.js

```diff
@@ -3,81 +3,83 @@ import * as Constants from '../../../constants/chat'
 import * as Creators from '../../../actions/chat/creators'
 import HiddenString from '../../../util/hidden-string'
 import ListComponent from '.'
-import {List, Map} from 'immutable'
+import {List} from 'immutable'
+import {compose} from 'recompose'
 import {connect} from 'react-redux'
+import {createSelector} from 'reselect'
 import {downloadFilePath} from '../../../util/file'
 import {navigateAppend} from '../../../actions/route-tree'
-import {pick} from 'lodash'
-import {compose} from 'recompose'
-import {getPath} from '../../../route-tree'
-import {chatTab} from '../../../constants/tabs'
 
 import type {OpenInFileUI} from '../../../constants/kbfs'
-import type {Options} from '../messages'
-import type {Props, OptionsFn} from '.'
 import type {OwnProps, StateProps, DispatchProps} from './container'
+import type {Props} from '.'
 import type {TypedState} from '../../../constants/reducer'
 
+// TODO change this. This is a temporary store for the messages so we can have a function to map from
+// messageKey to Message to support the 'edit last message' functionality. We should change how this works
+let _messages: List<Constants.Message> = List()
```

Comment:
we can change this is there's a simpler idea. right now when you press up to edit it has to do a bunch of crap. go find the actual last item (could use a saga?), then find the last dom node, then show an editor on top. This is slightly different than how mobile works. that sends an action and marks a message as isEditing cause it just puts a blue border around it and lets you edit it in the input box. lets likely leave this for now and change it later